### PR TITLE
feat(tabset): add animation

### PR DESCRIPTION
**Pull request by mattlewis92**
Preview:

 - [x] read and followed the [CONTRIBUTING.md](https://github.com/ng-bootstrap/ng-bootstrap/blob/master/CONTRIBUTING.md) guide.
 - [x] built and tested the changes locally.
 - [x] added/updated any applicable tests.
 - [x] added/updated any applicable API documentation.
 - [x] added/updated any applicable demos.

### CODE REVIEW

File: src/tabset/tabset.ts [hunk 1]
```diff
@@ -11,6 +11,8 @@ import {
   EventEmitter
 } from '@angular/core';
 import {NgbTabsetConfig} from './tabset-config';
+import {transition, trigger, useAnimation} from '@angular/animations';
+import {fadeIn, fadeOut} from '../animations/fade';
```
Comment (pkozlowski-opensource) [inline src/tabset/tabset.ts]: Unused import (`fadeOut`)

File: src/tabset/tabset.spec.ts [hunk 2]
```diff
@@ -165,7 +168,7 @@ describe('ngb-tabset', () => {
     const tabTitles = getTabTitles(fixture.nativeElement);
 
     expect(tabTitles[0].textContent).toMatch(/foo/);
-    expect(tabTitles[1].innerHTML).toMatch(/<b>bar<\/b>/);
+    expect(tabTitles[1].innerHTML).toMatch(/<b( class=""|)>bar<\/b>/);
```
Comment (pkozlowski-opensource) [inline src/tabset/tabset.spec.ts]: Any insights on why this tests needs changing? I'm not sure why, all of the sudden, this element is being touched... We even don't have animations on titles :-/

File: src/tabset/tabset.spec.ts [hunk 3]
```diff
@@ -213,24 +216,26 @@ describe('ngb-tabset', () => {
   });
 
 
-  it('should change active tab on tab title click', () => {
-    const fixture = createTestComponent(`
+  it('should change active tab on tab title click', fakeAsync(() => {
```
Comment (pkozlowski-opensource) [inline src/tabset/tabset.spec.ts]: Why this change? We are not testing for tabs content here and the test passes even without `fakeAsync`.

File: src/tabset/tabset.spec.ts (hunk 2 above)
Comment (mattlewis92) [reply src/tabset/tabset.spec.ts]: I believe its the angular animations module adding pseudo classes to elements, as on browsers in saucelabs that don't support have the web animations library the class wasn't added.

File: src/tabset/tabset.spec.ts [hunk 4]
```diff
@@ -368,8 +373,8 @@ describe('ngb-tabset', () => {
   });
 
 
-  it('should change active tab by calling select on an exported directive instance', () => {
-    const fixture = createTestComponent(`
+  it('should change active tab by calling select on an exported directive instance', fakeAsync(() => {
```
Comment (pkozlowski-opensource) [inline src/tabset/tabset.spec.ts]: Same question about `fakeAsync` - this test works for me without your change.

File: src/tabset/tabset.spec.ts (hunk 3 above)
Comment (mattlewis92) [reply src/tabset/tabset.spec.ts]: I just added the fakeAsync / flush to every test that clicked headers, I guess it can be removed on this one.

File: src/tabset/tabset.spec.ts [hunk 5]
```diff
@@ -378,58 +383,63 @@ describe('ngb-tabset', () => {
...
+       fixture.detectChanges();
+       flush();
+       let tabContents = getTabContent(fixture.nativeElement);
+       expect(tabContents.length).toBe(2);
+       expect(tabContents[1]).toHaveCssClass('active');
+     }));
 
-  it('should emit tab change event when switching tabs', () => {
-    const fixture = createTestComponent(`
+  it('should emit tab change event when switching tabs', fakeAsync(() => {
```
Comment (pkozlowski-opensource) [inline src/tabset/tabset.spec.ts]: Works without `fakeAsync` for me

**Comment by pkozlowski-opensource** [comment]
@mattlewis92 reviewed this one. The change in the code looks good but I would like to clarify what is going on inside tests. As noted, those are passing for me without `fakeAsync` so I'm puzzled by your changes. Care to elaborate?

Also, I'm not sure if the animations would ran on the initial tab display (it should not).

**Comment by mattlewis92** [comment]
re the tests I was in auto pilot mode and just added the fake async / flush to all tests that clicked the header, obviously it was not required for all of them and I'll remove the ones that don't need it.

Good shout on not running them on the initial tab display, I will get back to you on that.

**Comment by mattlewis92** [comment]
@pkozlowski-opensource I've addressed the changes you requested, this is ready for review again 😄

**Comment by pkozlowski-opensource** [comment]
Thnx for the update @mattlewis92 - it looks _almost_ perfect :-) The only thing I've noticed while playing with the demo is that now changing a tab makes the whole page (actually, content of demos) jump up and down for a split second.

I can reproduce it on Chrome 60.0.3112.113 (OSX) but not in other browsers (FFox, Chrome canary, Safari) so it is super-wired :-/ Did you see anything similar on your end?

I'm going to tests with some more browsers later today on a different machine and will let you know.

**Comment by mattlewis92** [comment]
I can reproduce it in chrome 60 as well, given that its not in canary though I'd say its probably a bug with the web-animations API that looks like its been fixed now.

**Comment by pkozlowski-opensource** [comment]
> I can reproduce it in chrome 60 as well, given that its not in canary though I'd say its probably a bug with the web-animations API that looks like its been fixed now.

Yep. I've just tested it with Chrome 61 and everything works perfectly fine. I'm merging this PR, thnx!

**Comment by pkozlowski-opensource** [comment]
Merged as a10183a037136f08b38992629ea0ecea41ab5291 to the https://github.com/ng-bootstrap/ng-bootstrap/tree/animations branch. Thnx!

**Comment by mattlewis92** [comment]
Ah, that was a well timed release 😄 Thanks for reviewing, LMK if there is anything else I can do to help with getting these PRs landed.

**Comment by pkozlowski-opensource** [comment]
> LMK if there is anything else I can do to help with getting these PRs landed.

No, it is fine, I'm just moving slowly as I'm learning animations with your PRs so it takes time :-)
But don't worry, we are on track for landing them all - and thnx for your excellent work on this!

And hey, don't let pending reviews stop you from sending more PRs :-)
